ga4gh/ga2vcf.py

```diff
@@ -0,0 +1,345 @@
+
+    # see pysam source file cvcf.pyx
+    numberTypes = {
+        'NT_UNKNOWN': 0,
+        'NT_NUMBER': 1,
+        'NT_ALLELES': 2,
+        'NT_NR_ALLELES': 3,
+        'NT_GENOTYPES': 4,
+        'NT_PHASED_GENOTYPES': 5,
+    }
+
+    def __init__(self, infoDict):
+        self.infoDict = infoDict
+        for key, value in self.infoDict.items():
+            setattr(self, key, value)
+
+    def getNumbertype(self):
+        # The google api seems to return one of the following for number:
+        # - an integer
+        # - a period
+        # - None (only on custom header lines)
+        if re.match('[0-9]+', self.number) is not None:
+            return self.numberTypes['NT_NUMBER']
+        elif self.number == 'R':
+            return self.numberTypes['NT_ALLELES']
+        elif self.number == 'A':
+            return self.numberTypes['NT_NR_ALLELES']
+        elif self.number == 'G':
+            return self.numberTypes['NT_GENOTYPES']
+        else:
+            return self.numberTypes['NT_UNKNOWN']
+
+    def __getattr__(self, name):
+        if name == 'numbertype':
+            return self.getNumbertype()
+
+
+class VcfMetadata(object):
+
+    def __init__(self):
+        self.info = {}
+        self.filter_ = {}
+        self.format_ = {}
+        self.header = []
+        self.switch = {
+            'INFO': self.info,
+            'FILTER': self.filter_,
+            'FORMAT': self.format_,
+        }
+
+    def processVariantSet(self, variantSet):
+        for metadata in variantSet.metadata:
+            infoLine = self._processMetadata(metadata)
+            if infoLine.key in self.switch:
+                self.switch[infoLine.key][infoLine.id] = infoLine
+            elif infoLine.key == 'fileformat':
+                # fileformat already written by pysam
+                continue
+            else:
+                keyVal = self._processHeader(infoLine)
+                self.header.append(keyVal)
+
+    def _processHeader(self, infoLine):
+        if infoLine.key == 'contig':
+            valueStr = []
+            for ilKey, ilVal in infoLine.info.items():
+                ilVal = ilVal[0]
+                if ilVal.find(' ') != -1:
+                    ilVal = '"{0}"'.format(ilVal)
+                fragment = '{0}={1}'.format(ilKey, ilVal)
+                valueStr.append(fragment)
+            joinedValueStr = ','.join(valueStr)
+            value = "<{0}>".format(joinedValueStr)
+        elif infoLine.key == 'ALT':
+            value = "<ID={id}, Description={description}>".format(
+                **infoLine.infoDict)
+        # TODO special cases for other fields: assembly, etc...
+        # see VCF doc section 1.2
+        else:
+            value = infoLine.value
+        return (infoLine.key, value)
+
+    def _processMetadata(self, metadata):
+        infoDict = metadata.toJsonDict()
+        infoLine = VcfMetadataInfoLine(infoDict)
+        return infoLine
+
+
+class VcfRow(object):
```

Comment:
Perhaps, but I'll worry about performance later... conversion to a function is an easy change.  And (see the other comment about getattr) there might be nasty surprises in pysam that mandate this... we'll see.  Plus, I like the encapsulation.